lib/route.js

```diff
@@ -142,6 +142,15 @@ function buildRouting (options) {
       throw new Error(`Error Handler for ${opts.method}:${opts.url} route, if defined, must be a function`)
     }
 
+    // Handle optional parameter
+    const path = opts.url || opts.path
+    const optionalParam = path.match(/:([^/]*)\?\/?$/)
+    if (optionalParam) {
+      opts.url = path.replace(/(:[^/]*)\?(\/?)$/, '$1$2')
+      const optionalPath = path.replace(/\/:([^/]*)\?(\/?)$/, '$2')
```

Comment:
These regular expressions need more explanation in the code. There's way too many brackets and optionals for these to be easily legible.